developer/src/kmc-model-info/src/model-info-compiler.ts

```diff
@@ -1,163 +1,209 @@
-  if(model_info.packageFileSize === undefined) {
-    callbacks.reportMessage(ModelInfoCompilerMessages.Error_FileDoesNotExist({filename:options.kmpFileName}));
-    return null;
-  }
+    // description
 
-  setModelMetadata('jsFilename', callbacks.path.basename(options.modelFileName));
+    if(sources.kmpJsonData.info.description?.description) {
+      model_info.description = sources.kmpJsonData.info.description.description.trim();
+    }
 
-  // Always overwrite with actual file size
-  model_info.jsFileSize = callbacks.fileSize(options.modelFileName);
-  if(model_info.jsFileSize === undefined) {
-    callbacks.reportMessage(ModelInfoCompilerMessages.Error_FileDoesNotExist({filename:options.modelFileName}));
-    return null;
-  }
+    // isRTL -- this is a little bit of a heuristic from a compiled .js
+    //          which may need modification if compilers change
 
-  // Always overwrite source data
-  model_info.packageIncludes = options.kmpJsonData.files.filter((e) => !!e.name.match(/.[ot]tf$/i)).length ? ['fonts'] : [];
+    if(jsFile?.match(/("?)isRTL("?):\s*true/)) {
+      model_info.isRTL = true;
+    }
+
+    // extract the language identifiers from the language metadata
+    // arrays for each of the lexical models in the kmp.json file,
+    // and merge into a single array of identifiers in the
+    // .model_info file.
+
+    model_info.languages = sources.kmpJsonData.lexicalModels.reduce((a, e) => [].concat(a, e.languages.map((f) => f.id)), []);
 
-  setModelMetadata('version', options.kmpJsonData.info.version.description);
+    // If a last commit date is not given, then just use the current time
+    model_info.lastModifiedDate = sources.lastCommitDate ?? (new Date).toISOString();
+
+    model_info.packageFilename = this.callbacks.path.basename(sources.kmpFileName);
+    model_info.packageFileSize = this.callbacks.fileSize(sources.kmpFileName);
+    if(model_info.packageFileSize === undefined) {
+      this.callbacks.reportMessage(ModelInfoCompilerMessages.Error_FileDoesNotExist({filename:sources.kmpFileName}));
+      return null;
+    }
 
-  // The minimum Keyman version detected in the package file may be manually set higher by the developer
-  setModelMetadata('minKeymanVersion', minKeymanVersion, false);
+    model_info.jsFilename = this.callbacks.path.basename(sources.modelFileName);
+    model_info.jsFileSize = this.callbacks.fileSize(sources.modelFileName);
+    if(model_info.jsFileSize === undefined) {
+      this.callbacks.reportMessage(ModelInfoCompilerMessages.Error_FileDoesNotExist({filename:sources.modelFileName}));
+      return null;
+    }
+
+    model_info.packageIncludes = sources.kmpJsonData.files.filter((e) => !!e.name.match(/.[ot]tf$/i)).length ? ['fonts'] : [];
+    model_info.version = sources.kmpJsonData.info.version.description;
+    model_info.minKeymanVersion = minKeymanVersion;
+    model_info.helpLink = HelpRoot + model_info.id;
```

Comment:
Are we going to be publishing lexical-model help? I didn't see HelpRoot in the help.keyman.com repo
https://help.keyman.com/model

Reply:
This leaves the option open for the future, but no current plans :grin: